### PR TITLE
[FIX] base_import: fix the import speed with many batches

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
Since v15, the time to import records in Odoo is much slower than in
v14. With a profiling, we know that the cause is `fields_get`. Indeed,
this method is heavy and call many times (#rows * #columns * ~#batches).

So the import time (in seconds) is;

```
Batch size | v14 | v15 | v15 fix
--------------------------------
2000       |   7 |  20 |      14
 200       |  11 |  48 |      15
  20       |  17 | 423 |      23
```

We can see that just by calling `._fields.get` instead of `fields_get`
we gain a lot of time. The reason for that is `fields_get` just read in
a python dictionary (which is really really fast), while `fields_get`
checks the access right, get the description of each fields (which
might make SQL queries to retrieve translation, etc).

Task-2687407

### CODE REVIEW

**Comment by Bonainifo** [comment]
Didn’t solve the fundamental problem? After the repair, it is still twice as slow as the 14 version?

**Comment by Bonainifo** [comment]
Didn’t solve the fundamental problem? After the repair, it is still twice as slow as the 14 version?

**Comment by std-odoo** [comment]
@Bonainifo the reported issue in the pad is not the real problem, it was already in v14.
It solve the issue because the time will not increase exponentially anymore, we created a other task because many other thing can be improved, but it required a refactor

**Comment by celm1990** [comment]
sorry for comment, can review `flush ` method and records `m2o ` are imported one by one and not in batch, see https://github.com/odoo/odoo/pull/54290

File: addons/base_import/models/base_import.py [hunk 1]
```diff
@@ -1424,7 +1424,8 @@ def _handle_multi_mapping(self, import_fields, input_file_data):
                 for field in split_fields:
                     if field != target_field:  # if not on the last hierarchy level, retarget the model
                         target_model = self.env[target_model][field]._name
-                field_type = self.env[target_model].fields_get().get(target_field, {}).get('type', '')
+                field = self.env[target_model]._fields.get(target_field)
```
Comment (tde-banana-odoo) [inline addons/base_import/models/base_import.py:1427]: MessageRemark! First commit: missing explanations (import time on which kind of records ? what is the number of records to import ? what are globally the installed apps) ?

Also missing an explanation of your fix (fields_get is slow -> ok, and then ?) .

File: addons/base_import/tests/test_base_import.py [hunk 2]
```diff
@@ -707,6 +707,57 @@ def test_skip_record_import(self):
         # if results empty, no errors
         self.assertItemsEqual(results['messages'], [])
 
+    def test_multi_mapping(self):
```
Comment (tde-banana-odoo) [inline addons/base_import/tests/test_base_import.py:710]: Ok for the test, and no fix related to it ? "This commit is done as a preliminary step to then improve the performance of
the method that handles the multi-mapping." -> is it planned for this PR ? Of master only ? missing some context here :)

**Review by tde-banana-odoo — COMMENTED** [review]
Had a look :)

**Comment by std-odoo** [comment]
@tde-banana-odoo I updated the commit messages :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**